cosmo_tester/test_suites/cluster/full_cluster_test.py

```diff
@@ -65,6 +68,59 @@ def test_full_cluster_names(full_cluster_names, logger, ssh_key, test_config):
     check_managers(mgr1, mgr2, example)
 
 
+def test_cluster_5_0_5_snapshot_with_idd(full_cluster_ips, logger, ssh_key,
+                                         test_config):
+    broker1, broker2, broker3, db1, db2, db3, mgr1, mgr2 = full_cluster_ips
+
+    snapshot_id = 'snap_5.0.5_with_component_openstack'
+    _upload_snapshot_from_url(mgr2, logger, snapshot_id)
+    restore_snapshot(mgr2, snapshot_id, logger, force=True,
+                     cert_path=mgr2.local_ca)
+
+    deployment_ids = [d.id for d in mgr1.client.deployments.list()]
+    assert {'getting-started.mc-jboss', 'infra-openstack'} == \
+           set(deployment_ids)
+
+    with pytest.raises(CloudifyClientError) as e:
+        mgr1.client.deployments.delete('infra-openstack')
+    assert "Can't delete deployment infra-openstack" in str(e.value)
+    assert "`getting-started.mc-jboss` uses capabilities" in str(e.value)
+    assert "`getting-started.mc-jboss` contains the current deployment" in str(
+        e.value)
+
+    with pytest.raises(CloudifyClientError) as e:
+        mgr1.client.executions.start('infra-openstack', 'uninstall')
+    assert "Can't execute workflow `uninstall` on deployment " \
+           "infra-openstack" in str(e.value)
+    assert "`getting-started.mc-jboss` uses capabilities" in str(e.value)
+    assert "`getting-started.mc-jboss` contains the current deployment" in str(
+        e.value)
+
+
+def test_cluster_4_6_0_snapshot_with_idd(full_cluster_ips, logger, ssh_key,
+                                         test_config):
+    broker1, broker2, broker3, db1, db2, db3, mgr1, mgr2 = full_cluster_ips
+
+    snapshot_id = 'snap_4.6.0_with_capabilities'
+    _upload_snapshot_from_url(mgr2, logger, snapshot_id)
+    restore_snapshot(mgr2, snapshot_id, logger, force=True,
+                     cert_path=mgr2.local_ca)
+
+    deployment_ids = [d.id for d in mgr1.client.deployments.list()]
+    assert {'infra', 'capable'} == set(deployment_ids)
+
+    with pytest.raises(CloudifyClientError) as e:
+        mgr1.client.deployments.delete('infra')
+    assert "Can't delete deployment infra" in str(e.value)
+    assert "`capable` uses capabilities" in str(e.value)
+
+    with pytest.raises(CloudifyClientError) as e:
+        mgr1.client.executions.start('infra', 'uninstall')
+    assert "Can't execute workflow `uninstall` on deployment " \
+           "infra" in str(e.value)
+    assert "`capable` uses capabilities" in str(e.value)
```

Comment:
Aside from the snapshot name and the content of the error messages, these look to be basically doing the same thing, so maybe they could be factored out to make it easier to maintain if we change the test.